### PR TITLE
Remove lines from test_clear_cache

### PR DESCRIPTION
**Pull request by kfollesdal**
Was a little fast yesterday and discover that `assert c.__dict__ == fresh` did not do what I intended. Is only checking keys and not values of the dictionary. But think the test is good as it is now. 

Thanks, for the tips with nox. Have run nox and did get a problem with isort in lint. Have made some changes to import to pass nox -s lint. But main reason is probably the line ending since I am sitting on a Windows computer.

### CODE REVIEW

File: omegaconf/basecontainer.py [hunk 1]
```diff
@@ -256,7 +256,7 @@ def pretty(self, resolve: bool = False, sort_keys: bool = False) -> str:
     @staticmethod
     def _map_merge(dest: "BaseContainer", src: "BaseContainer") -> None:
         """merge src into dest and return a new copy, does not modified input"""
-        from omegaconf import OmegaConf
+        from omegaconf import OmegaConf  # isort:skip
```
Comment (omry) [inline omegaconf/basecontainer.py]: I never had to do this.
what problem is this solving?

File: tests/test_interpolation.py [hunk 2]
```diff
@@ -288,9 +288,7 @@ def test_clear_cache(restore_resolvers: Any) -> None:
     OmegaConf.register_resolver("random", lambda _: random.randint(0, 10000000))
     c = OmegaConf.create(dict(k="${random:_}"))
     old = c.k
-    fresh = c.__dict__
     OmegaConf.clear_cache(c)
-    assert c.__dict__ == fresh
```
Comment (omry) [inline tests/test_interpolation.py]: the new code looks better and is not using undocumented behavior.

File: omegaconf/basecontainer.py (hunk 1 above)
Comment (kfollesdal) [reply omegaconf/basecontainer.py]: These lines was failing when I run `isort . --check` and isort did not manage to fix them. But the common factor is that they are imports inside methods. Buy adding `# isort:skip`, isort passed with out errors.

Comment (kfollesdal) [reply omegaconf/basecontainer.py]: Do nox pass at your system `#isort:skip` ? Do you have a more correct way to do this?

Comment (kfollesdal) [reply omegaconf/basecontainer.py]: The pull request contains the last changes from master. So no rebase is necessary.

Comment (omry) [reply omegaconf/basecontainer.py]: nox is passing for me on master.
does it not pass for you?

Show me the output of `nox -s lint-3.6` on master for you.

Comment (kfollesdal) [reply omegaconf/basecontainer.py]:
```
nox > Running session lint-3.6
nox > Creating virtual environment (virtualenv) using python.exe in .nox\lint-3-6
nox > pip install --upgrade setuptools pip
nox > pip install -r requirements/lint.txt -e .
nox > mypy . --strict
nox > isort . --check
nox > Command isort . --check failed with exit code 1:
ERROR: C:\Users\kf9762\Desktop\tmp\omegaconf\omegaconf\base.py Imports are incorrectly sorted.
ERROR: C:\Users\kf9762\Desktop\tmp\omegaconf\omegaconf\basecontainer.py Imports are incorrectly sorted.
nox > Session lint-3.6 failed.
```

Comment (omry) [reply omegaconf/basecontainer.py]: Seems to work for me:
```
omry@omry-fedora:~/dev/omegaconf (master)$ git describe --tags
2.0.0rc6-41-gcffc694
omry@omry-fedora:~/dev/omegaconf (master)$ nox -s lint-3.6
nox > Running session lint-3.6
nox > Creating virtual environment (virtualenv) using python3.6 in .nox/lint-3-6
nox > pip install --upgrade setuptools pip
nox > pip install -r requirements/lint.txt -e .
nox > mypy . --strict
nox > isort . --check
nox > black --check .
nox > Session lint-3.6 was successful.
```

**Comment by omry** [comment]
On Windows:
```
omry@omry-PC MINGW64 ~/PycharmProjects/omegaconf (master)
$ git describe --tags
2.0.0rc6-41-gcffc694
omry@omry-PC MINGW64 ~/PycharmProjects/omegaconf (master)
$ nox -s lint-3.6
Running session lint-3.6
Creating virtual environment (virtualenv) using python.exe in .nox\lint-3-6
pip install --upgrade setuptools pip
pip install -r requirements/lint.txt -e .
mypy . --strict
isort . --check
black --check .
Session lint-3.6 was successful.

```

**Comment by kfollesdal** [comment]
Strange. Have to be something with my system. Good that it works on your system an the pipeline. Did a completely new setup project on my system. And still, get errors. Got some more error now but that is LF/CRLF problem. 

```
(venv) λ git describe --tags
2.0.0rc6-41-gcffc694

(venv) λ nox -s lint-3.6
nox > Running session lint-3.6
nox > Creating virtual environment (virtualenv) using python.exe in .nox\lint-3-6
nox > pip install --upgrade setuptools pip
nox > pip install -r requirements/lint.txt -e .
nox > mypy . --strict
nox > isort . --check
nox > Command isort . --check failed with exit code 1:
ERROR: C:\Users\kf9762\Desktop\tmp\test\omegaconf\omegaconf\base.py Imports are incorrectly sorted.
ERROR: C:\Users\kf9762\Desktop\tmp\test\omegaconf\omegaconf\basecontainer.py Imports are incorrectly sorted.
ERROR: C:\Users\kf9762\Desktop\tmp\test\omegaconf\omegaconf\dictconfig.py Imports are incorrectly sorted.
ERROR: C:\Users\kf9762\Desktop\tmp\test\omegaconf\omegaconf\listconfig.py Imports are incorrectly sorted.
ERROR: C:\Users\kf9762\Desktop\tmp\test\omegaconf\omegaconf\nodes.py Imports are incorrectly sorted.
ERROR: C:\Users\kf9762\Desktop\tmp\test\omegaconf\omegaconf\omegaconf.py Imports are incorrectly sorted.
ERROR: C:\Users\kf9762\Desktop\tmp\test\omegaconf\omegaconf\_utils.py Imports are incorrectly sorted.
ERROR: C:\Users\kf9762\Desktop\tmp\test\omegaconf\tests\test_basic_ops_dict.py Imports are incorrectly sorted.
ERROR: C:\Users\kf9762\Desktop\tmp\test\omegaconf\tests\test_nodes.py Imports are incorrectly sorted.
ERROR: C:\Users\kf9762\Desktop\tmp\test\omegaconf\tests\test_serialization.py Imports are incorrectly sorted.
ERROR: C:\Users\kf9762\Desktop\tmp\test\omegaconf\tests\test_utils.py Imports are incorrectly sorted.
ERROR: C:\Users\kf9762\Desktop\tmp\test\omegaconf\tests\structured_conf\test_structured_config.py Imports are incorrectly sorted.
nox > Session lint-3.6 failed.
```

**Comment by omry** [comment]
Okay, please remove all the `# isort:skip` you added and push again to see if CI likes it.
(I fixed the CI issue that causes your previous PR to not trigger it).

You can also try dos2unix locally to fix the newlines.

**Comment by omry** [comment]
Thanks!

**Comment by omry** [comment]
By the way, I released rc16 earlier.

**Pull request merged**